src/data_generators/m3_data_design.py

```diff
@@ -157,19 +166,200 @@
             "order": OrderOptions.random,
             "correlated_sets": CorrelatedSetsOptions.one,
             "shared_prop": 0.75,
-            "random_generator": 8,
+            "random_generator": np.random.default_rng(RNG.integers(1e9)),
         },
     ),
+]
+
+
+class CopulaCorrelationMatrixGenerator:
+    @staticmethod
+    def homogeneous(p: int, rho: float) -> np.ndarray:
+        """Generate a homogeneous correlation matrix.
+
+        Args:
+            p:  Number of pubs.
+            rho:  Homogenous correlation.
+
+        Returns:
+            A p * p matrix with diagonals being 1 and all off-diagonals being rho.
+        """
+        return splinalg.toeplitz([1] + [rho] * (p - 1))
+
+    @staticmethod
+    def autoregressive(p: int, rho: float) -> np.ndarray:
+        """Generate a autoregressive correlation matrix.
+
+        Args:
+            p:  Number of pubs.
+            rho:  Homogenous correlation.
+
+        Returns:
+            A <p * p> matrix C where C[i, j] = rho^|i - j| for any i, j.
+        """
+        return splinalg.toeplitz(np.power(np.array([rho] * p), np.arange(p)))
+
+    @staticmethod
+    def random(
```

Comment:
Is there a unit test for this?

Reply:
There was a basic test which checked if the generated matrices are positive definite and thus a valid correlation matrix.  I will add another test to check the random-uniformity at least when p = 2, after confirming the definition of uniformity with Chenwei.